### DISCUSSION TITLE
pytest markers not being picked up under class test

### DISCUSSION THREAD

**Discussion by Formartha** [pytest-dev/pytest#12621] [General]
Hi,

I have a class test and setup, teardown methods and also tests in it. however when marking the test with a marker it fail to fetch it and deselect all the tests including the marked on. What might be wrong there?

``` import logging
import pytest

from ***** import BaseManagementManualUpgradeTest


logger = logging.getLogger(__name__)


class TestManualManagementUpgrade(BaseManagementManualUpgradeTest):
    """
    manual upgrade test
    """

    @classmethod
    def setup_class(cls: BaseManagementManualUpgradeTest):
        some_code_that_was_deleted

    @pytest.mark.sanity
    def test_manual_master_upgrade(self):
        self.run_manual_upgrade()
```

**Comment by RonnyPfannschmidt** [comment]
with the information provided we cant even start to guess
please try to make a minimal reproducible example 

include the base classes
the pytest invocation

**Reply by Formartha** [reply]
@RonnyPfannschmidt I can't share much, but looking at the item (Iv'e used the item collection hook and accessed the item) it seems like no marker is being registered on the node (i.e. the test)

![image](https://github.com/user-attachments/assets/667f2fcc-7e50-49f1-8442-3aada216a906)

**Reply by Formartha** [reply]
I found the issue, it seems like if there are two tests under two different test classes and different path, it still consider them as a single test, hence, It can't collect it.

Is this behavior expected?

**Reply by webknjaz** [reply]
@Formartha you should be able to come up with a [SSCCE](https://sscce.org) to demonstrate the behavior without sharing any private code. Then, people would be able to give answers based on what they see and not guesses. But it sounds like this behavior wouldn't be expected, based on my fortune-telling skills...

**Reply by Formartha** [reply]
@webknjaz , I understand but the question is general as most of the infrastructure is encapsulated (if you'd ask me, it's incorrect, but that's the case and I'm here to try to solve some of the issues).
Currently there is the same test method and same test names under different directories. The directories are under 'tests' root folder which is used to store all the test cases.

What happens is that for some reason, one of the tests are being ignored and some aren't, I'm unsure why, as when I put the markers I want to run under different test, it does run.

I don't see any exceptions on collections so I imagine it's some sort of issue with the project layout or defect in the collection structure.

**Reply by RonnyPfannschmidt** [reply]
As no details on invocation and structure was provided we can't provide and help for lack of starting points

**Reply by Formartha** [reply]
ok, so here is a way to try to reproduce it:
1. create a directory and name it `tests/`
2. create two folders (with `__init__.py`) in it and name them `build/` and `example/`
3. create two exact test files and name them `test_1.py`
4. paste the given example in it and try to run it from the `tests/` directory.
```python
# tests/test_1.py
class TestOne():

    def test_1_exp(self):
        assert 1 == 1
```

Only after changing the name of the `build/` folder to `build_1/` it works… I'm not sure why (and it's not what I thought at the beginning, just tried to mimic it and found the issue by mistake)

**Reply by RonnyPfannschmidt** [reply]
I suspect this relates to collect ignore trying to ignore the setuptools build folder 

Which would be a bug for non build folders

**Reply by Formartha** [reply]
Should we raise a defect here or this is enough information?

**Reply by webknjaz** [reply]
I suppose a bug report could ask for documenting this explicitly… `build/` is often managed by `setuptools` so it's expected that it won't be a part of an actual project structure.

**Reply by RonnyPfannschmidt** [reply]
we ought to only consider a build directory as ignored if its next to a pyprojct.toml/setup.cfg or setup.py

**Reply by RonnyPfannschmidt** [reply]
i created #12625
